### PR TITLE
[data] [docs] Basic docs for Data Train integration

### LINKED ISSUE #32605: [data] [streaming] Support streaming splitted datasets in AIR
As part of enabling streaming datasets by default in Ray 2.4, we also need to make it usable in AIR. The main blocker here is that AIR sometimes uses the .split() API for Ray Train, which currently forces a fallback to bulk execution mode.

We need to:
- Implement a new streaming_split() API that can be used to consume the streaming output from multiple workers.
- Integrate and test with Train.

Note: replacing DatasetPipeline entirely with streaming execution is out of scope of this issue (that will be post-2.4).

### PR DESCRIPTION
**Pull request by ericl**
## Why are these changes needed?

Followup for https://github.com/ray-project/ray/pull/35236

This only contains the main doc page. The example and reference updates are to be done in https://github.com/ray-project/ray/issues/35958

## Related issue number

Closes https://github.com/ray-project/ray/issues/32605

### CODE REVIEW

File: doc/source/ray-air/check-ingest.rst [hunk 1]
```diff
@@ -3,444 +3,171 @@
...
+        dataset_config=MyCustomDataConfig(),
+    )
+    my_trainer.fit()
+
+
+What do you need to know about this ``DataConfig`` class?
+
+* It must be serializable, since it will be copied from the driver script to the driving actor of the Trainer.
+* Its ``configure`` method is called on the main actor of the Trainer group to setup data ingest and iterators.
+* Train will get data shards for each worker by calling ``get_shard`` on the instance after setup is called.
```
Comment (raulchen) [inline doc/source/ray-air/check-ingest.rst:171]: Is this already implemented? I didn't find this `get_shard` method in #35236

File: doc/source/ray-air/check-ingest.rst [hunk 2]
```diff
@@ -3,444 +3,171 @@
...
+          iterator_shards = datasets["train"].streaming_split(
+               world_size,
+               equal=True,
+               locality_hints=worker_node_ids)
+
+          # Create shards for each worker and return.
+          outputs = {}
+          for rank in range(world_size):
+              outputs[rank] = {"train": iterator_shards[rank]}
+          self.output_splits = outputs
```
Comment (raulchen) [inline doc/source/ray-air/check-ingest.rst:148]: Nit, if this method is intended to be overridden by subclasses. Better make it side-effect free. I.e., just "return outputs", not changing "self.output_splits".

File: doc/source/ray-air/check-ingest.rst (hunk 1 above)
Comment (ericl) [reply doc/source/ray-air/check-ingest.rst:171]: Let me move to doc_code.

**Comment by ericl** [comment]
Will move more stuff to doc_code.

**Comment by ericl** [comment]
@raulchen examples should be working now, sorry about that.

**Review by raulchen — APPROVED** [review]
LGTM

**Review by richardliaw — APPROVED** [review]

**Pull request merged**